hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Normalization.scala

```diff
@@ -249,7 +290,17 @@ object Normalization:
     case (Lit(Tree.StrLit(_)), ClassLike(blt.`Str`, _, _, _)) => true
     case (Lit(Tree.DecLit(_)), ClassLike(blt.`Num`, _, _, _)) => true
     case (Lit(Tree.BoolLit(_)), ClassLike(blt.`Bool`, _, _, _)) => true
-    case (_, _) => false
+    case (_: Synonym, _: Synonym) => lhs =:= rhs
+    case (Record(entries1), Record(entries2)) =>
+      entries1.forall { (fieldName1, _) => entries2.exists { (fieldName2, _) => fieldName1 === fieldName2 } }
+    case (Record(entries), ClassLike(cs: ClassSymbol, _, _, _)) =>
+      val clsParams = cs.defn match
+        case S(Parameterized(params = paramList)) => paramList.params
+        case (S(_)|N) => List()
```

Comment:
`Nil` is more efficient

```suggestion
        case (S(_) | N) => Nil
```